include/caf/caf_overview.rst

```diff
@@ -0,0 +1,186 @@
+
+Enabling CAF
+************
+
+To enable CAF, set the :option:`CONFIG_CAF` Kconfig option, initialize Event Manager, and submit the first :c:struct:`module_state_event`.
+You can then enable each :ref:`CAF module <lib_caf>` separately.
+
+Initializing Event Manager
+==========================
+
+As CAF uses Event Manager, you must initialize Event Manager in your :file:`main.c` file.
+To do so, complete the following steps:
+
+1. Include :file:`event_manager.h`.
+#. Call :c:func:`event_manager_init()`.
+
+Submitting ``module_state_event``
+=================================
+
+CAF modules are initialized when they receive a :c:struct:`module_state_event` that reports ``MODULE_STATE_READY`` of the module ``main``.
+For this to happen, you need to do the following:
+
+1. Add the following in your :file:`main.c` file:
+
+.. code-block:: c
+
+   #define MODULE main
+   #include <caf/events/module_state_event.h>
+
+#. Call the following function after Event Manager is initialized:
+
+.. code-block:: c
+
+   module_set_state(MODULE_STATE_READY);
+
+This function call submits the required :c:struct:`module_state_event`.
+For an example implementation, see :file:`applications/nrf_desktop/src/main.c`.
+
+CAF modules
+***********
+
+CAF supports a set of :ref:`predefined modules <lib_caf>`.
+These modules are available for use out of the box.
+You only need to enable and configure them.
+
+There is no limit to the number of modules and events that can be used.
+However, Event Manager sets the limit of event types in an application to 32, and CAF inherits this limitation.
+
+Configuration of CAF modules
+============================
+
+Each module must be configured separately.
+When configuring CAF modules, you will be asked to use the following configuration methods:
+
+* Edit Devicetree Specification (DTS) files - These files reflect the hardware configuration.
+  See :ref:`zephyr:dt-guide` for more information about the DTS data structure.
+* Edit :file:`_def` files - These files contain configuration arrays for the application modules and are specific to CAF.
+* Select Kconfig options or modify Kconfig files, or both - These reflect the software configuration.
+  See :ref:`kconfig_tips_and_tricks` for information about how to configure them.
+
+CAF modules can use all or some of these methods.
+For example, enabling the :ref:`caf_ble_adv` requires enabling Kconfig options and creating a configuration file, while :ref:`caf_ble_state` can be configured using only Kconfig options.
+
+Adding custom modules
+*********************
+
+As part of your application, you can add your own custom modules.
+These modules can communicate using Event Manager's events.
+However, if you want to add custom modules that communicate with CAF modules using CAF events, you must make sure that they meet the following requirements:
+
+* A custom module must subscribe for Event Manager's events.
```

Comment:
`CAF events`?
Otherwise it's a bit unclear when we read it together with previous sentence

Reply:
We may add information that you can subscribe for a CAF event in the same way as you subscribe for any other Event Manager event